remotes/resolver.go

```diff
@@ -50,12 +51,19 @@ type Resolver interface {
 	Pusher(ctx context.Context, ref string) (Pusher, error)
 }
 
-// Fetcher fetches content
+// Fetcher fetches content.
+// A fetcher implementation may implement the FetcherByDigest interface too.
 type Fetcher interface {
 	// Fetch the resource identified by the descriptor.
 	Fetch(ctx context.Context, desc ocispec.Descriptor) (io.ReadCloser, error)
 }
 
+// FetcherByDigest fetches content by the digest.
+type FetcherByDigest interface {
+	// FetchByDigest the resource identified by the digest.
+	FetchByDigest(ctx context.Context, dgst digest.Digest) (io.ReadCloser, int64, error)
```

Comment:
Returning `ocispec.Descriptor` would be useful here over just `int64`. The media type could always be `application/octet-stream` since it was not fetched with any known/trusted context. Annotations could also be set to include the media type returned by the registry. Generally though we use the Descriptor type for blob metadata, I don't lacking a media types makes that less useful here.

Reply:
Updated to return the desc with `application/octet-stream`